From 789b9b46ee1b0f4f84cd5186e557a12378b16fe5 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Sat, 17 Jun 2017 00:49:49 +0800
Subject: [PATCH 1952/5242] MLK-15074-9 usb: gadget: utp: fix wrong fsg
 parameter

commit  daad0fe32494b2968efdb1e74526a8a4091f5d83 from
https://source.codeaurora.org/external/imx/linux-imx.git

As common->fsg maybe have not been set correctly while enumration, so
use the correct pointer fsg for utp device check.

Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/gadget/function/f_mass_storage.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/function/f_mass_storage.c b/drivers/usb/gadget/function/f_mass_storage.c
index 57123cd..2d03e56 100644
--- a/drivers/usb/gadget/function/f_mass_storage.c
+++ b/drivers/usb/gadget/function/f_mass_storage.c
@@ -2998,7 +2998,7 @@ static int fsg_bind(struct usb_configuration *c, struct usb_function *f)
 	fsg->interface_number = i;
 
 #ifdef CONFIG_FSL_UTP
-	if (is_utp_device(common->fsg))
+	if (is_utp_device(fsg))
 		utp_init(fsg);
 #endif
 
-- 
1.7.9.5

